files/usr/share/cinnamon/cinnamon-settings/bin/Spices.py

```diff
@@ -392,7 +393,14 @@ def _url_retrieve(self, url, outfd, reporthook, binary):
             proxy = proxygsettings.get_proxy_settings()
             if proxy and proxy.get('https'):
                 connection = HTTPSConnection(proxy.get('https'), timeout=15)
-                connection.set_tunnel(host)
+                if proxy and proxy.get('http'):
+                    proxy_url = urlparse("//" + proxy.get('http'))
+                    if proxy_url.username and proxy_url.password:
+                        proxy_auth = '%s:%s' % (proxy_url.username, proxy_url.password)
+                        headers = {"Proxy-Authorization": "Basic " + str(b64encode(proxy_auth.encode())).replace("b'", "").replace("'", "")}
```

Comment:
Wouldn't this be simpler?
```python
'Basic %s' % (b64encode(proxy_auth.encode()).decode())
```

Reply:
Yes, definitely simpler. Python is not my native language ;)
Seems I've made a mess with commits here, sorry.